pkgs/development/libraries/catboost/default.nix

```diff
@@ -0,0 +1,114 @@
+{ lib
+, config
+, stdenv
+, fetchFromGitHub
+, cmake
+, libiconv
+, llvmPackages
+, ninja
+, openssl
+, python3
+, ragel
+, yasm
+, zlib
+, cudaSupport ? config.cudaSupport or false
+, cudaPackages ? {}
+, enablePython ? false
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "catboost";
+  version = "1.2";
+
+  src = fetchFromGitHub {
+    owner = "catboost";
+    repo = "catboost";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-Ng1ylqL9fBRJP5hATv9FzJTqqSIFEUN0At1DJ3V2FM4=";
+  };
+
+  patches = [
+    ./remove-conan.patch
+    ./fix-imported-targets.patch
+  ] ++ lib.optionals (lib.versionOlder cudaPackages.cudatoolkit.version "11.8") [
+    ./remove-unsupport-cc.patch
+  ];
+
+  postPatch = ''
+    substituteInPlace cmake/common.cmake \
+      --replace  "\''${RAGEL_BIN}" "${lib.getExe ragel}" \
+      --replace "\''${YASM_BIN}" "${lib.getExe yasm}"
+  '';
+
+  outputs = [
+    # binary and libraries e.g. bin/catboost, lib/catboost.so, etc.
+    "out"
+    # libraries required for Python, R, etc. support
+    # see https://catboost.ai/en/docs/installation/build-native-artifacts#targets
+    "dev"
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    llvmPackages.bintools
+    ninja
+    ragel
+    yasm
+  ];
+
+  buildInputs = [
+    openssl
+    zlib
+  ] ++ lib.optionals stdenv.isDarwin [
+    libiconv
+  ] ++ lib.optionals cudaSupport (with cudaPackages; [
+    cuda_cccl
+    cuda_cudart
+    cuda_nvcc
+  ]);
+
+  env = {
+    CUDAHOSTCXX = "${stdenv.cc}/bin/cc";
+    NIX_CFLAGS_LINK = lib.optionalString stdenv.isLinux "-fuse-ld=lld";
+    NIX_LDFLAGS = "-lc -lm";
+    # protoc uses this variable
+    LD_LIBRARY_PATH = "${lib.getLib zlib}/lib";
```

Comment:
Can we use patchelf instead? LD_LIBRARY_PATH is rather dirty

Reply:
I didn't know that using patchelf would solve this.
I added an autoPatchelfHook. Thanks.

Reply:
In the hindsight, I'm a bit confused: how exactly does `protoc` use LD_LIBRARY_PATH, and what exactly is it that autoPatchelfHook ends up patchelfing? How does the bulid fail when both autopatchelf and ld_library_path are omitted?

Reply:
The build should have failed because protoc could not find zlib. 
I don't know if it was due to a version bump or my mistake, but this hack was no longer needed.